tests/nn/data_parallel/test_sharded_ddp_pytorch_parity.py

```diff
@@ -201,28 +196,30 @@ def closure_sharded(input_tensor=input_tensor):
 @pytest.mark.parametrize("reduce_buffer_size", [0, 2 ** 20])
 @pytest.mark.parametrize("grad_accumulation", [True, False])
 @pytest.mark.parametrize("change_train_graph", [True, False])
-@pytest.mark.parametrize("fp16_reduction", _test_fp16_reduction)
-def test_ddp_parity(reduce_buffer_size, grad_accumulation, change_train_graph, fp16_reduction):
+@pytest.mark.parametrize("clip_grad_norm", [True, False])
```

Comment:
two new cases, which showed the problem that @zhengwy888 had spotted: check clip_grad_norm parity, with and without AMP, without and without buckets, with and without grad accumulation, and change in training graph. This reliably showed an error when the buckets were activated

Reply:
![Screen Shot 2021-03-16 at 10 53 09 AM](https://user-images.githubusercontent.com/3072861/111554802-43cd8380-8744-11eb-93fa-daa8a4db58b3.png)

Reply:
checked in with another PR